linux_kernel_modules/rtc_sync/rtc_sync.c

```diff
@@ -100,30 +102,32 @@ static int systohc(struct timespec now)
 	if (rtc) {
 		/* rtc_hctosys exclusively uses UTC, so we call set_time here,
 		 * not set_mmss. */
-		if (rtc->ops &&
-		    (rtc->ops->set_time ||
-		     rtc->ops->set_mmss))
+		if (rtc->ops && (rtc->ops->set_time || rtc->ops->set_mmss))
 			err = rtc_set_time(rtc, &tm);
-		rtc_class_close(rtc);
+			rtc_class_close(rtc);
 	}
-        
 	return 0;
 }
 
-//time_check chekcs if the system time has been modified during the last checking period by comparing the expected time after checking period with a time allowance
+/*
+ * time_check chekcs if the system time has been modified during 
+ * the last checking period by comparing the expected time after 
+ * checking period with a time allowance static int time_check(void *data)
+ */
+
 static int time_check(void *data)
```

Comment:
I rewrote this function in a way that I think makes it a bit clearer and shorter.  I haven't tested this, but I think it's correct.  What do you think?
```
static int time_check(void *data)
{
	while (!kthread_should_stop()) {
		const long long expected_delta_ns = 10 * NSEC_PER_SEC;
		long timeout = 10 * HZ;
		struct timespec before;
		struct timespec after;
		long long delta_ns;

		ktime_get_real_ts(&before);
		while (timeout > 0)
			timeout = schedule_timeout(timeout);
		ktime_get_real_ts(&after);

		delta_ns = timespec_to_ns(&after) - timespec_to_ns(&before);
		if (abs64(delta_ns - expected_delta_ns) > 1 * NSEC_PER_SEC)
			systohc(after);
	}
	return 0;
}

```